src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/StopApprenticeship/StopApprenticeshipCommandHandler.cs

```diff
@@ -0,0 +1,110 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.Authentication;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Exceptions;
+using SFA.DAS.CommitmentsV2.Messages.Commands;
+using SFA.DAS.CommitmentsV2.Messages.Events;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.Encoding;
+using SFA.DAS.NServiceBus.Services;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Application.Commands.StopApprenticeship
+{
+    public class StopApprenticeshipCommandHandler : AsyncRequestHandler<StopApprenticeshipCommand>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ICurrentDateTime _currentDate;
+        private readonly IAuthenticationService _authenticationService;
+        private readonly IEventPublisher _eventPublisher;
+        private readonly IMessageSession _nserviceBusContext;
+        private readonly IEncodingService _encodingService;
+        private readonly ILogger<StopApprenticeshipCommandHandler> _logger;
+        private const string StopNotificationEmailTemplate = "ProviderApprenticeshipStopNotification";
+
+        public StopApprenticeshipCommandHandler(
+            Lazy<ProviderCommitmentsDbContext> dbContext,
+            ICurrentDateTime currentDate,
+            IEventPublisher eventPublisher,
+            IAuthenticationService authenticationService,
+            IMessageSession nserviceBusContext,
+            IEncodingService encodingService,
+            ILogger<StopApprenticeshipCommandHandler> logger)
+        {
+            _dbContext = dbContext;
+            _currentDate = currentDate;
+            _eventPublisher = eventPublisher;
+            _authenticationService = authenticationService;
+            _nserviceBusContext = nserviceBusContext;
+            _encodingService = encodingService;
+            _logger = logger;
+        }
+
+        protected async override Task Handle(StopApprenticeshipCommand command, CancellationToken cancellationToken)
```

Comment:
I think this should be `override async`, and `command` should be named `request` to match the interface definition (pedantic, I know, and probably not consistently applied at all).

Reply:
Have renamed 👍